### PR TITLE
Fix `selector-type-*`, `selector-max-type` and `selector-class-pattern` false positives for `<timeline-range-name>` in keyframe selectors

### PR DESCRIPTION
**Pull request by romainmenke**
> Which issue, if any, is this issue related to?

Closes https://github.com/stylelint/stylelint/issues/7855

> Is there anything in the PR that needs further explanation?

This is a blocker for https://github.com/stylelint/stylelint/pull/7856#discussion_r1682845978.

### CODE REVIEW

File: .changeset/afraid-crabs-ring.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: `selector-*` false positives for `timeline-range-name` in `keyframe` selectors
```
Comment (Mouvedia) [inline .changeset/afraid-crabs-ring.md:5]:
```suggestion
Fixed: `selector-type-*`, `selector-max-type` and `selector-class-pattern` false positives for `<timeline-range-name>` in keyframe selectors
```

File: lib/reference/keywords.mjs [hunk 2]
```diff
@@ -286,6 +286,16 @@ export const camelCaseKeywords = new Set([
 
 export const keyframeSelectorKeywords = new Set(['from', 'to']);
 
+// https://drafts.csswg.org/scroll-animations-1/#view-progress-timelines
+export const namedTimelineRangeKeywords = new Set([
```
Comment (Mouvedia) [inline lib/reference/keywords.mjs:290]: Can you export this set in `keyframe-selector-notation/index.mjs` and replace

https://github.com/stylelint/stylelint/blob/e7e54941b9376bc63d5c7343c1d1ef56eb0ab04c/lib/rules/keyframe-selector-notation/index.mjs#L20

Comment (romainmenke) [reply lib/reference/keywords.mjs:290]: I think we best do that in a separate PR.

The list used by `keyframe-selector-notation` is either wrong or outdated.
`enter` does not exist (anymore) and `entry-crossing`, `exit-crossing` do exist.

Fixing that is out of scope.

Comment (Mouvedia) [reply lib/reference/keywords.mjs:290]: We will have to wait for this PR to be merged then.

File: lib/utils/isKeyframeSelector.mjs [hunk 3]
```diff
@@ -1,4 +1,9 @@
-import { keyframeSelectorKeywords } from '../reference/keywords.mjs';
+import { keyframeSelectorKeywords, namedTimelineRangeKeywords } from '../reference/keywords.mjs';
+
+const HAS_TIMELINE_RANGE = new RegExp(
+	`^(?:${[...namedTimelineRangeKeywords.values()].join('|')})\\s+(?:\\d+|\\d*\\.\\d+)%$`,
```
Comment (Mouvedia) [inline lib/utils/isKeyframeSelector.mjs:4]: trivial
note

I wonder if browser support unusual `<number>`s.
e.g. `10e3%`, `-3.4e-2%`

Comment (romainmenke) [reply lib/utils/isKeyframeSelector.mjs:4]: Out of scope?

I doubt that anyone writes CSS likes this If anyone does, then they can open an issue and let us know :)

Comment (Mouvedia) [inline lib/utils/isKeyframeSelector.mjs:4]: note

This requires the selector to be trimmed before being passed.

File: .changeset/afraid-crabs-ring.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: `selector-type-*`, `selector-max-type` and `selector-class-pattern` false positives for `<timeline-range-name>` in keyframe selectors
```
Comment (Mouvedia) [inline .changeset/afraid-crabs-ring.md:5]: question
minor

Does it underline the percentage too?
i.e. Should it be "…for `<timeline-range-name> <percentage>` in…"?

**Review by Mouvedia — APPROVED** [review]

File: lib/utils/isKeyframeSelector.mjs (hunk 3 above)
Comment (romainmenke) [reply lib/utils/isKeyframeSelector.mjs:4]: Correct, like the other checks in this function

File: .changeset/afraid-crabs-ring.md (hunk 4 above)
Comment (romainmenke) [reply .changeset/afraid-crabs-ring.md:5]: I guess it depends on the rule?
I think the current text is descriptive enough to communicate what was fixed.

**Review by ybiquitous — APPROVED** [review]
@romainmenke Thanks. LGTM 👍🏼 

As a bonus, it might be great to add a test case to each rule affected by this PR, so that users might understand this bug fix more easily. Of course, this is not required, though. We can also merge this as-is. 😃

**Comment by romainmenke** [comment]
I've added some extra tests, also good to have those independent of the helper :)
Thank you

**Pull request merged**